db/versions/0001.yml

```diff
@@ -6,6 +6,11 @@ migrationScript: |-
     );
     grant select, insert, update, delete on widgets to $db_user_prefix$_notify;
   end
+downgradeScript: |-
+  begin
+    revoke select, insert, update, delete on widgets from $db_user_prefix$_notify;
+    drop table widgets;
```

Comment:
Would it be sensible to have a test that makes sure the number of grants/revokes in the migration script is equal to the number of revokes/grants?

Reply:
I think a more reliable thing might be to validate that after downgrading to version 0, all grants are gone.  I'll see if I can cook that up.